docroot/modules/custom/mass_entity_usage/mass_entity_usage.install

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * @file
+ * Installation functions for Mass Entity Usage.
+ */
+
+use Drupal\Core\Database\Database;
+
+/**
+ * Add index to paragraphs_item_field_data on id and revision id.
+ */
+function mass_entity_usage_update_8106() {
+  $spec = [
+    'fields' => [
+      'id' => [
```

Comment:
I'm a little surprised to see these. Aren't there already indexes on this, since it's a primary key? And I would have expected `id, revision_id` to be a key too. If not, should we file an issue upstream?

Reply:
Right, those 2 keys are primary keys by themself, but I'm creating a new index based on those 2 keys.
Not sure if we have to post a fix/patch in the issue upstream.
AFAIK an index means you get faster reads, but slower inserts.

Reply:
Ooo, I think I realized why this index isn't in core.

The revision ID is already a **unique key**. `\Drupal\Core\Entity\EntityStorageInterface::loadRevision()` takes just a revision ID, not an entity ID. In other words, adding the primary key doesn't to the query doesn't actually do anything! The only way you'd need this index is if revision IDs were not unique and started from `1` for each entity.

Reply:
Lets use our existing bundle classes for now, and skip Typed Entity. We waited this long to intro bundle classes because we want to stick to vanilla core in this case.

Reply:
@weitzman @beto-aveiga what do you think about my comment on the unique key?

Reply:
@deviantintegral interesting! I think you are correct. Which also means that this condition can be removed...  https://github.com/massgov/openmass/pull/1338/files#diff-d04531c7b9896c1f1c59a10b38ae00d4c957267c33912dc7d7a41697dc31423fR63  Right?

Reply:
Yes, I think so!